### PR TITLE
feat: add Translatable annotation

### PR DESCRIPTION
**Pull request by vietnguyen**
https://jira.dhis2.org/browse/DHIS2-10260
Introduce new annotation `@Translatable`

After this is merged, the steps for enable translation for a property should be 
1) Add a method `getDisplay<propertyName>()` to the object model
2) Add the annotation `@Translatable` to the `getDisplay*` method

A `translatable` property is also added to `Property` class and exposed the endpoint /api/schema. So each properties in the response will have two attributes related to translation :

```
"translationKey":"NAME"
"translatable":true
```

### CODE REVIEW

**Comment by netroms** [comment]
Mini typo: 
> response will have **to** > Two? attributes related to translation :

**Review by netroms — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-schema/src/test/java/org/hisp/dhis/schema/TranslatablePropertyIntrospectorTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.schema;/*
+                             * Copyright (c) 2004-2020, University of Oslo
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-schema/src/test/java/org/hisp/dhis/schema/TranslatablePropertyIntrospectorTest.java:29]: Remove license.

**Comment by mortenoh** [comment]
Please merge with master and apply formatting

**Comment by vietnguyen** [comment]
> Please merge with master and apply formatting

Done

File: dhis-2/dhis-services/dhis-service-schema/src/test/java/org/hisp/dhis/schema/TranslatablePropertyIntrospectorTest.java (hunk 1 above)
Comment (vietnguyen) [reply dhis-2/dhis-services/dhis-service-schema/src/test/java/org/hisp/dhis/schema/TranslatablePropertyIntrospectorTest.java]: fixed

**Comment by larshelge** [comment]
I was wondering if we could just infer the translation key from the get-method based on naming convention. The naming convention is:

`getDisplay{PropertyName}`

E.g. if the get-method is called `getDisplayShortName()`, we can infer the property to be `shortName`.

It would also be interesting to re-use the @ Property annotation with a `translateable = true` argument, instead of having a new annotation. Not sure if that is possible/feasible.

**Review by netroms — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/validation/ValidationRule.java [hunk 2]
```diff
@@ -233,9 +233,10 @@ public String getInstruction()
 
     @JsonProperty
     @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    @Translatable( propertyName = "instruction", key = "NSTRUCTION" )
```
Comment (mortenoh) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/validation/ValidationRule.java:236]: Should be `INSTRUCTION` ?

**Review by mortenoh — CHANGES_REQUESTED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/validation/ValidationRule.java (hunk 2 above)
Comment (vietnguyen) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/validation/ValidationRule.java]: fixed

**Review by mortenoh — APPROVED** [review]

**Pull request merged**